app/models/miq_request_task.rb

```diff
@@ -140,12 +140,12 @@ def deliver_queue(req_type = request_type, zone = nil)
 
     _log.info("Queuing #{request_class::TASK_DESCRIPTION}: [#{description}]...")
 
-    if resource_action&.configuration_script_payload
-      workflow = resource_action.configuration_script_payload
+    workflow = ConfigurationScriptPayload.find(options[:configuration_script_payload_id]) if options[:configuration_script_payload_id]
+    if workflow
```

Comment:
find will raise ActiveRecord::RecordNotFound if the id doesn't exist, so I'm not sure how workflow could be falsey on line 144.

Reply:
yup - I ignored the `if` clause at the end.